packages/components/carousel/src/carousel.vue

```diff
@@ -300,17 +315,25 @@ watch(
   }
 )
 
+watch(
+  () => items.value.map((item) => item.uid),
+  (ids, oldIds) => {
+    if (activeIndex.value < 0) setActiveItem(props.initialIndex)
+    else {
+      const activeId = oldIds[activeIndex.value]
+      const index = ids.indexOf(activeId)
+      if (index > -1) setActiveItem(index)
+      else setActiveItem(activeIndex.value)
+    }
+  }
+)
+
```

Comment:
I think when the items change, we don't need to maintain the active state, but directly reset the active index to the initial index. This behavior is consistent with `element-ui`.

```ts
if (ids.length > 0) setActiveItem(props.initialIndex)
```